daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -320,10 +324,12 @@ object SExpr {
           machine.env.size + bounds.size - 1))
 
       // Start evaluating the let binders
-      for (i <- 1 until bounds.size) {
+      var i = 0
```

Comment:
```suggestion
      var i = 1
```

Reply:
Ah, that's why the tests were failing. Good catch. Thank you.